main.go

```diff
@@ -108,31 +109,79 @@ func main() {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
+
+	certsReady := make(chan struct{})
+	if err = kueuev1alpha1.ManageCerts(mgr, certsReady); err != nil {
+		setupLog.Error(err, "unable to set up cert rotation")
+		os.Exit(1)
+	}
+
+	cCache := cache.New(mgr.GetClient())
+	queues := queue.NewManager(mgr.GetClient(), cCache)
+
+	if err := setupIndexes(mgr); err != nil {
+		setupLog.Error(err, "Unable to setup indexes")
+	}
+
+	setupProbeEndpoints(mgr, certsReady)
+	go startControllers(mgr, cCache, queues, certsReady, config.ManageJobsWithoutQueueName)
```

Comment:
why in a routine?

Reply:
Cert won't be ready until manager starts, add a comment here. 
Controllers can be added after manager starts and will started immediately. See here: 
https://github.com/kubernetes-sigs/controller-runtime/blob/0c2effbc7eabc502eb79472b6c6ba8fbb6ec8b76/pkg/manager/runnable_group.go#L49-L69

Reply:
I see. Does the cert controller need to run before the certs can be ready?